### PR TITLE
Collapse IonDotnet.Tree

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: IonDotnet/Tree/IIonValue.cs [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ﻿using System.Collections.Generic;
-using IonDotnet.Tree.Impl;
 
 namespace IonDotnet.Tree
 {
-    public interface IIonValue
+    public interface IIonValue : IIonInt, IIonFloat, IIonTimestamp, IIonDecimal,
```
Comment (simonz-bq) [inline IonDotnet/Tree/IIonValue.cs]: I think this list should be ordered in the same way that the binaryId is.

File: IonDotnet/Tree/Impl/IonValue.cs [hunk 2]
```diff
@@ -352,6 +354,34 @@ public void WriteTo(IIonWriter writer)
         /// <summary>The <see cref="IonType"/> of this value.</summary>
         public abstract IonType Type { get; }
 
+        // Applicable to IonDecimal
```
Comment (simonz-bq) [inline IonDotnet/Tree/Impl/IonValue.cs:362]: NotImplementedException probably isn't right here.

I would have a const or static string for the message, saying something like "This operation is not supported for IonType this.Type". Then, maybe an InvalidOperationException which would throw the exception with the message.

**Review by simonz-bq — CHANGES_REQUESTED** [review]

File: IonDotnet/Tree/IIonValue.cs (hunk 1 above)
Comment (ghost) [reply IonDotnet/Tree/IIonValue.cs]: Done.

File: IonDotnet/Tree/Impl/IonValue.cs (hunk 2 above)
Comment (ghost) [reply IonDotnet/Tree/Impl/IonValue.cs:362]: Done.

File: IonDotnet/Tree/IIonValue.cs [hunk 3]
```diff
@@ -14,9 +15,7 @@ public interface IIonValue
         string ToPrettyString();
         void WriteTo(IIonWriter writer);
         bool IsEquivalentTo(IIonValue value);
-        IonType Type { get; }
```
Comment (simonz-bq) [inline IonDotnet/Tree/IIonValue.cs:17]: These 2 functions need to be exposed in the IIonValue API

**Pull request closed without merging**